icevision/utils/model_creation.py

```diff
@@ -0,0 +1,232 @@
+
+    Parameters
+    ----------
+    backend_lib_name: str
+        Name of the backend_lib to get the backbones for
+    model_type_name: str
+        Name of the model type to get the backbones for
+    base_path: str
+        Path from where to read out the available models from
+
+    Returns
+    -------
+    backbones: List[str]
+        List of available backbones
+    """
+    print(backend_lib_name)
+    print(model_type_name)
+    backbone_files = [
+        i
+        for i in os.listdir(
+            os.path.join(base_path, backend_lib_name, model_type_name, "backbones")
+        )
+        if os.path.join(
+            base_path, backend_lib_name, model_type_name, "backbones", i
+        ).endswith(".py")
+        and not i.startswith("__")
+    ]
+    backbone_list = []
+    for backbone_file in backbone_files:
+        backbones = importlib.import_module(
+            "icevision.models."
+            + ".".join([backend_lib_name, model_type_name])
+            + ".backbones."
+            + backbone_file.replace(".py", "")
+        )
+        backbone_list += getattr(backbones, "__all__", [])
+    return backbone_list
+
+
+def load_model_components(
+    backend_lib_name: str,
+    model_type_name: str,
+    backbone_name: str,
+    model_base_path: str = BASE_PATH,
+):
+    """Build a model form given backend lib, model_type and backbone.
+
+    Parameters
+    ----------
+    backend_lib_name: str
+        Name of the backend_lib to use
+    model_type_name: str
+        Name of the model type to use
+    backbone_name: str
+        Name of the backbone to use
+
+    Returns
+    -------
+    model_type: Module
+        model_type to be used to create the dataloaders and so on
+    backbone: object
+        backbone for the model
+    """
+    # load the backend lib
+    try:
+        backend_lib = get_module_element_form_module(models, backend_lib_name)
+    except AttributeError:
+        raise AttributeError(
+            f"Backend lib: {backend_lib_name} not found. Possible options are: {get_backend_libs(model_base_path)}"
+        )
+    # load the model_type
+    try:
+        model_type = get_module_element_form_module(
+            models, backend_lib_name, model_type_name
+        )
+    except AttributeError:
+        raise AttributeError(
+            f"Model type: {model_type_name} not found. Possible options are: {get_model_types_for_backend_lib(model_base_path, backend_lib_name)}"
+        )
+    # load backbone
+    try:
+        backbone_module = get_module_element_form_module(
+            models, backend_lib_name, model_type_name, "backbones"
+        )
+        backbone = getattr(backbone_module, backbone_name)
+    except AttributeError:
+        raise AttributeError(
+            f"Backbone: {backbone_name} not found. Possible options are: {get_backbone_names(backend_lib_name, model_type_name)}"
+        )
+
+    return model_type, backbone
+
+
+def build_model(
+    backend_lib_name: str,
```

Comment:
Wdyt about simplifying the argument name by using:
`lib_name`, `model_name` instead of  `backend_lib_name` and  `model_type_name`

Also, would it be better to pass a dictionary instead of argument. Maybe using a dict, we will loose auto-completion.